pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -671,6 +690,22 @@ func GetShortNameMode(ctx *types.SystemContext) (types.ShortNameMode, error) {
 // (This is split from the caller primarily to make testing easier.)
 func refMatchesPrefix(ref, prefix string) bool {
 	switch {
+	case prefix[0:2] == "*.":
+		if strings.Contains(ref, "/") {
+			firstSlash := strings.Index(ref, "/")
```

Comment:
The search for `/` should be possible to do only once: `strings.Contains == (strings.Index() != -1)` IIRC.

Also elsewhere.